airflow/contrib/operators/mysql_to_gcs.py

```diff
@@ -221,19 +218,42 @@ def _upload_to_gcs(self, files_to_upload):
         for object, tmp_file_handle in files_to_upload.items():
             hook.upload(self.bucket, object, tmp_file_handle.name, 'application/json')
 
-    @classmethod
-    def convert_types(cls, value):
+    def _convert_types(self, schema, col_type_dict, row):
         """
         Takes a value from MySQLdb, and converts it to a value that's safe for
         JSON/Google cloud storage/BigQuery. Dates are converted to UTC seconds.
-        Decimals are converted to floats.
+        Decimals are converted to floats. Binary type fields are encoded with base64,
+        as imported BYTES data must be base64-encoded according to Bigquery SQL
+        date type documentation: https://cloud.google.com/bigquery/data-types
+        """
+        converted_row = []
+        for col_name, col_val in zip(schema, row):
+            if type(col_val) in (datetime, date):
+                col_val = time.mktime(col_val.timetuple())
+            elif isinstance(col_val, Decimal):
+                col_val = float(col_val)
+            elif col_type_dict.get(col_name) == "BYTES":
+                col_val = base64.urlsafe_b64encode(col_val)
+                if PY3:
+                    col_val = col_val.decode('ascii')
+            else:
+                col_val = col_val
+            converted_row.append(col_val)
+        return converted_row
+
+    def _get_col_type_dict(self):
+        """
+        Return a dict of column name and column type based on self.schema if not None.
         """
-        if type(value) in (datetime, date):
-            return time.mktime(value.timetuple())
-        elif isinstance(value, Decimal):
-            return float(value)
+        schema = []
+        if isinstance(self.schema, string_types):
+            schema = json.loads(self.schema)
+        elif isinstance(self.schema, list):
+            schema = self.schema
         else:
-            return value
+            return {}
```

Comment:
Maybe add a warning here:

```
elif self.schema is not none:
    self.log.warn("Ignoring self.schema due to unexpected type. Should be a string or list.")
```

you don't really need the else case

Reply:
got it